valhalla/loki/polygon_search.h

```diff
@@ -0,0 +1,67 @@
+#ifndef VALHALLA_LOKI_POLYGON_SEARCH_H_
+#define VALHALLA_LOKI_POLYGON_SEARCH_H_
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/geometry/geometries/register/ring.hpp>
+
+#include <valhalla/baldr/graphreader.h>
+#include <valhalla/midgard/constants.h>
+#include <valhalla/midgard/pointll.h>
+#include <valhalla/proto/options.pb.h>
+#include <valhalla/sif/dynamiccost.h>
+
+#include <functional>
+
+// TODO: register PointLL in its own header, so it's available for bg across valhalla?
+BOOST_GEOMETRY_REGISTER_POINT_2D(valhalla::midgard::PointLL,
+                                 double,
+                                 boost::geometry::cs::geographic<boost::geometry::degree>,
+                                 first,
+                                 second)
```

Comment:
you wanna keep it contained here or put the registration into `PointLL` header?

Reply:
i would move as much as i can out of the header and into the source file (polygon_search.cc). ive not tried it but it would be great if it all could be sequestered to there.. same with line_bg_t and ring_bg_t. i havent tried this myself but it seems like the function below neither takes any of these as input nor returns them as output so you should be able to move them cleanly? im not exactly sure what the macro does but yeah put it in that single compilation unit would be the nicest way to manage it if it can still work that way